### ISSUE TITLE
FR: Add client side timeout notification

### ISSUE THREAD

**Issue by SteveRodrigue** [tailscale/tailscale#13447]
### What are you trying to do?

I (user) wants to see a notification on upcoming session timeout that requires a full re-auth.

### How should we solve this?

An OS or client notification should "pop". Something like:

> Your Tailscale session will timeout in 2 hours, do you want to re-authenticate now? [Yes/No]

Yes, should force an authentication process (expiring the current session).
So, we (users) would have control on when to authenticate.

Ideally, users should be able to define how much in advance they want to be notified. For instance, 12-24 hours would prevent almost any user to be disrupted during an important meeting, job, task, etc.

### What is the impact of not solving this?

At the moment, on Windows (client at least) there is no notification and the session just expires without any warning. This disrupt works.

### Anything else?

There's not even a Tailscale notification on disconnection, so we have to check the client status to make sure it's Tailscale that disconnected.

**Comment by agottardo** [comment]
@SteveRodrigue I'm told we do support notifications prior to key expiry on Windows. Did you disable notifications anywhere on your system?

**Comment by SteveRodrigue** [comment]
@agottardo : I don't think so and it would not fix the whole problem. We (users) should be able to set a reminder to X hours prior to timeout/reauth.

At the moment, it just drops the connection. If you have any tip(s) to check/assert the notification exists, it would be nice because the Tailscale UI or in the Web page personal settings I don't see anywhere settings related to notification(s) for Tailscale.

**Comment by agottardo** [comment]
cc @nickkhyl @dblohm7 anything we can do to determine why key expiry notifications aren't firing for this user on Windows?
